### PR TITLE
Fix: Table sorting doesn't work

### PR DESCRIPTION
**Pull request by cythrawll**
* Metric object was being passed where metricName was expected
* sort direction couldn't be found for dateTime sort
* Added sorting acceptance tests that would have caught this

### CODE REVIEW

File: packages/visualizations/addon/components/navi-visualizations/table.js [hunk 1]
```diff
@@ -203,7 +203,7 @@ export default Component.extend({
         columns = $.extend(true, [], get(this, 'options.columns'));
 
     return columns.map( column => {
-      let sort =  arr(sorts).findBy('metric', canonicalizeMetric(column.field)) || {};
+      let sort =  arr(sorts).findBy('metric', column.type === 'dateTime' ? column.type : canonicalizeMetric(column.field)) || {};
```
[the remark this answers is not in the document]
Comment (cythrawll) [reply packages/visualizations/addon/components/navi-visualizations/table.js]: Fixed

File: packages/visualizations/addon/components/navi-visualizations/table.js [hunk 2]
```diff
@@ -203,12 +203,13 @@ export default Component.extend({
         columns = $.extend(true, [], get(this, 'options.columns'));
 
     return columns.map( column => {
-      let sort =  arr(sorts).findBy('metric', canonicalizeMetric(column.field)) || {};
-      let sortDirection;
+      let {field, type} = column,
+          sort =  arr(sorts).findBy('metric', type === 'dateTime' ? type : canonicalizeMetric(field)) || {},
```
Comment (balajimanoharan) [inline packages/visualizations/addon/components/navi-visualizations/table.js]: Can we break this down some more? 
Extract ` type === 'dateTime' ? type : canonicalizeMetric(field)` into a different var maybe?
I feel there is too much happening in this line and a little hard to follow.

File: packages/reports/tests/acceptance/report-visualizations-test.js [hunk 3]
```diff
@@ -32,3 +32,44 @@ test('filter changes line chart series', function(assert) {
     });
   });
 });
+
+test('Table metric column sort', function(assert) {
+  assert.expect(11);
+  
+  visit('/reports/2/view');
+  andThen(() => {
+    assert.notOk(find('.table-header-cell.metric:first .navi-table-sort-icon').hasClass('navi-table-sort-icon--desc'));
```
Comment (balajimanoharan) [inline packages/reports/tests/acceptance/report-visualizations-test.js]: Can we add messages to these tests?

File: packages/visualizations/addon/components/navi-visualizations/table.js (hunk 2 above)
Comment (cythrawll) [reply packages/visualizations/addon/components/navi-visualizations/table.js]: Fixed

File: packages/reports/tests/acceptance/report-visualizations-test.js (hunk 3 above)
Comment (cythrawll) [reply packages/reports/tests/acceptance/report-visualizations-test.js]: Fixed

File: packages/reports/tests/acceptance/report-visualizations-test.js [hunk 4]
```diff
@@ -32,3 +32,55 @@ test('filter changes line chart series', function(assert) {
     });
   });
 });
+
+test('Table metric column sort', function(assert) {
```
Comment (balajimanoharan) [inline packages/reports/tests/acceptance/report-visualizations-test.js]: `Table metric/dateTime column sort`?

File: packages/reports/tests/acceptance/report-visualizations-test.js [hunk 5]
```diff
@@ -32,3 +32,55 @@ test('filter changes line chart series', function(assert) {
...
+      'Metric is not sorted asc after third sort click');
+
+    assert.ok(find('.table-header-cell.dateTime:first .navi-table-sort-icon').hasClass('navi-table-sort-icon--desc'),
+      'Datetime is sorted desc by default');
+    click('.table-header-cell.dateTime:first .navi-table-sort-icon');  
+  });
+
+  andThen(() => {
+    assert.ok(find('.table-header-cell.dateTime:first .navi-table-sort-icon').hasClass('navi-table-sort-icon--asc'),
+      'Datetime is sorted asc aftfer first sort click');
```
Comment (balajimanoharan) [inline packages/reports/tests/acceptance/report-visualizations-test.js]: `after` is misspelt
